### PR TITLE
Adds basic Content-Security-Policy header

### LINKED ISSUE #972: Implement CSP, says Moz Observatory
Quoth Mozilla Observatory, which [gives brew.sh](https://observatory.mozilla.org/analyze/brew.sh) a **D** rating as of [d60d598](https://github.com/Homebrew/brew.sh/commit/d60d598a608541d2e7bca0590f14853f85dc4e90):

Test | Pass | Score | Reason | Info
-- | -- | -- | -- | --
Content Security Policy | ❌ | -25 | Content Security Policy (CSP) header not implemented |  Content Security Policy (CSP) can prevent a wide range of cross-site scripting (XSS) and clickjacking attacks against your website.

Docs on CSP: <https://infosec.mozilla.org/guidelines/web_security#content-security-policy>

Looks like we might be able to achieve this with the simple addition of this to the first child node of `head`:

```html
<meta http-equiv="Content-Security-Policy" content="default-src https:">
```

This may take some finagling to ensure it's set right.

We cannot set the HTTP Headers since we're using GitHub Pages.

### PR DESCRIPTION
**Pull request by colindean**
This should adequately cover the basic CSP use case requiring https for all assets and disallow framing except within our own site.

Possible extensions include limiting loading to just self and cdn.jsdelivr.net with fetch of algolia.net, too. We'd probably also need images from avatars2.githubusercontent.com for blog posts.

Fixes #972

### CODE REVIEW

File: _layouts/base.html [hunk 1]
```diff
@@ -9,6 +9,7 @@
 <!DOCTYPE html>
 <html {% if page.direction == "rtl" %}dir="rtl" {% endif %}lang="{{ page.lang | default: "en" }}">
   <head>
+    <meta http-equiv="Content-Security-Policy" content="default-src https: ; frame-ancestors 'self'">
```
Comment (colindean) [inline _layouts/base.html:12]: Hard mode:

```suggestion
    <meta http-equiv="Content-Security-Policy"
          content="frame-ancestors 'self';
                    default-src 'none';
                    font-src https://fonts.gstatic.com;
                    img-src 'self' https://avatars2.githubusercontent.com https://avatars.githubusercontent.com;
                    object-src 'none';
                    script-src 'self' https://cdn.jsdelivr.net;
                    style-src 'self' https://cdn.jsdelivr.net;
                    require-trusted-types-for 'script'">
```

**Review by mistydemeo — APPROVED** [review]

**Review by SMillerDev — APPROVED** [review]

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks @colindean!

**Comment by MikeMcQuaid** [comment]
Site is broken with this change, reverting, sorry:

Screenshot 2023-08-21 at 14 44 42

**Comment by MikeMcQuaid** [comment]
Sorry, it seems this was not ready to be merged. Please make these PRs draft in future, thanks ❤️

**Pull request merged**